micro_sam/sam_annotator/_widgets.py

```diff
@@ -123,6 +123,58 @@ def _add_shape_param(self, names, values, min_val, max_val, step=1):
 
         return x_param, y_param, layout
 
+    #     # New method for directory selection
+    # def _add_directory_param(self, name, value, title=None):
+    #     layout = QtWidgets.QHBoxLayout()
+    #     layout.addWidget(QtWidgets.QLabel(name if title is None else title))
+
+    #     directory_textbox = QtWidgets.QLineEdit()
+    #     directory_textbox.setText(value)
+    #     layout.addWidget(directory_textbox)
+
+    #     directory_button = QtWidgets.QPushButton("Browse")
+    #     directory_button.clicked.connect(lambda: self._get_directory(name, directory_textbox))
+    #     layout.addWidget(directory_button)
+
+    #     return layout
+    def _add_directory_param(self, name, value, title=None, select_file=False):
```

Comment:
From what I understand this now works for directories and files (which is good!) I would then rename it to `_add_path_param`.